### PR TITLE
Re-enable cluster ingress operator (remove from overriden set)

### PR DESCRIPTION
**Pull request by ramr**
Drop openshift-cluster-ingress-operator from list of cvo-overrides so that
it is re-enabled as the ingress operator has changed to work out of the box.
Associated jira ticket: https://jira.coreos.com/browse/NE-88

And dropping as per comments in #415 

@ironcladlou @abhinavdahiya @rajatchopra  PTAL  Thx  - not sure who else to cc.

*Edited*  the whole shebang!

### CODE REVIEW

**Comment by ironcladlou** [comment]
Can we disable `tectonic-ingress-controller-operator` as part of this?

File: pkg/asset/manifests/content/bootkube/cvo-overrides.go [hunk 1]
```diff
@@ -35,10 +35,6 @@ overrides:
   namespace: openshift-cluster-network-operator
   name: cluster-network-operator
   unmanaged: true
-- kind: Deployment                    # this conflicts with tectonic-ingress-controller-operator
-  namespace: openshift-cluster-ingress-operator
-  name: cluster-ingress-operator
-  unmanaged: true
 - kind: ServiceAccount                # missing run level 0 on the namespace and has 0000_08
```
Comment (rajatchopra) [inline pkg/asset/manifests/content/bootkube/cvo-overrides.go]: Did you mean to drop both the deployment and service account? As per comments in #415?

Comment (ramr) [reply pkg/asset/manifests/content/bootkube/cvo-overrides.go]: no only the deployment - the service account is for the cluster-dns-operator.

**Comment by ramr** [comment]
@abhinavdahiya  done PTAL Thx
btw, the file name is too long for the  70 char subject line limit so truncated it to the directory name and changed the title accordingly ... also `git log|show` does have a `--stat` option!

@ironcladlou  I know a 3/4 files `pkg/asset/manifests/tectonic.go` and the actual asset content but am not sure what else is needed. Still checking thru' the code/repo. Thx

**Comment by abhinavdahiya** [comment]
@ramr 
> am not sure what else is needed

1. you can drop all `tectonic-ingress-*.go` from https://github.com/openshift/installer/tree/master/pkg/asset/manifests/content/tectonic
2. delte these line https://github.com/openshift/installer/blob/master/pkg/asset/manifests/tectonic.go#L65-L70 and https://github.com/openshift/installer/blob/master/pkg/asset/manifests/tectonic.go#L44-L49 and https://github.com/openshift/installer/blob/master/pkg/asset/manifests/tectonic.go#L53
for 1, 2 i'm not sure if the `tls` secret is no longer required?
3.  drop https://github.com/openshift/installer/blob/master/pkg/asset/manifests/content/bootkube/02-ingress-namespace.go
4. and delete this https://github.com/openshift/installer/blob/master/pkg/asset/manifests/operators.go#L209

This should be mostly the chunk of code.

**Comment by ramr** [comment]
@abhinavdahiya  thx  for the info. 
Did notice that `./pkg/asset/manifests/content/tectonic/tectonic-system-01-ca-cert.go` depends on the `IngressCaCert`?  So am assuming I'd need to keep whatever code that sets `IngressCaCert`.
Unless you think the system-ca-cert code is also a candidate to be removed?

And there's some cleanup to un{used,referenced} fields in  `tectonicTemplateData` in `pkg/asset/manifests/template.go`.

**Comment by abhinavdahiya** [comment]
lets remove `./pkg/asset/manifests/content/tectonic/tectonic-system-01-ca-cert.go` too :)

> And there's some cleanup to un{used,referenced} fields in tectonicTemplateData in pkg/asset/manifests/template.go

Yeah that would be nice too.

**Comment by ramr** [comment]
@abhinavdahiya  done and squashed the commits. PTAL Thx

**Comment by abhinavdahiya** [comment]
The changes look good. lets see if the ci passes. will `/lgtm` as soon as that happens.

**Comment by wking** [comment]
[e2e-aws][1]:

```
Waiting for router to be created ...
NAME                           STATUS    ROLES       AGE       VERSION
ip-10-0-147-178.ec2.internal   Ready     worker      1h        v1.11.0+d4cacc0
ip-10-0-147-211.ec2.internal   Ready     worker      1h        v1.11.0+d4cacc0
ip-10-0-158-155.ec2.internal   Ready     worker      1h        v1.11.0+d4cacc0
ip-10-0-18-133.ec2.internal    Ready     master      1h        v1.11.0+d4cacc0
ip-10-0-2-102.ec2.internal     Ready     master      1h        v1.11.0+d4cacc0
ip-10-0-47-156.ec2.internal    Ready     master      1h        v1.11.0+d4cacc0
ip-10-0-6-122.ec2.internal     Ready     bootstrap   1h        v1.11.0+d4cacc0
Waiting for router to be created ...
Another process exited
```

Sometimes those timeouts are just flakes.

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/467/pull-ci-openshift-installer-master-e2e-aws/734/build-log.txt

**Comment by abhinavdahiya** [comment]
testing locally
```
oc -n openshift-cluster-ingress-router get pods
NAME                   READY     STATUS    RESTARTS   AGE
router-default-pxqbp   1/1       Running   0          1h
```
i think it is necessary to run router in `openshift-ingress`. cc @smarterclayton

**Comment by ironcladlou** [comment]
@abhinavdahiya 
>i think it is necessary to run router in openshift-ingress. cc @smarterclayton 

Why?

@smarterclayton 
>Yeah, let’s us openshift-ingress as the namespace for the operator , less to type and cluster is redundant

Let's not, because we'll have to make a bunch of changes. Also, you seem to be talking about the operator and @abhinavdahiya is talking about the namespace for routers managed by the operator.

I don't want to change any of it, not right now. These namespaces aren't configurable and everything's tested as-is today.

**Comment by ironcladlou** [comment]
Okay, talked with @smarterclayton and he convinced me it's easier for us to change the operator than to change the other stuff that wants routers in `openshift-ingress`. Will reference PR shortly.

**Comment by abhinavdahiya** [comment]
@ironcladlou 
the origin-e2e care https://github.com/openshift/origin/blob/01b0649c894ffb88a6f1a2d0a41d3abfa3f0b80c/test/extended/util/framework.go#L1429-L1452
the ci cares https://github.com/openshift/release/blob/46bde58076e277fe5c9f8afaab9d922ad74b3df8/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L129-L142

**Comment by ironcladlou** [comment]
Operator and router namespaces renamed in https://github.com/openshift/cluster-ingress-operator/pull/52.

**Comment by ironcladlou** [comment]
@abhinavdahiya @smarterclayton @ramr 
>@ironcladlou
the origin-e2e care https://github.com/openshift/origin/blob/01b0649c894ffb88a6f1a2d0a41d3abfa3f0b80c/test/extended/util/framework.go#L1429-L1452
the ci cares https://github.com/openshift/release/blob/46bde58076e277fe5c9f8afaab9d922ad74b3df8/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L129-L142

Both of those are going to have to change, as the new default router is actually a `daemonset` named `router-default`.

**Comment by ironcladlou** [comment]
@smarterclayton @abhinavdahiya @deads2k can someone explain why we even need this router gate at all? https://github.com/openshift/release/pull/1952 Could we just delete the gating logic?

**Comment by wking** [comment]
> ... can someone explain why we even need this router gate at all?

We want to wait until the cluster is "up" before running the e2e tests.  I guess the router is just a good marker for that.

**Comment by ironcladlou** [comment]
>We want to wait until the cluster is "up" before running the e2e tests. I guess the router is just a good marker for that.

Wouldn't the apiserver be a better indicator in the abstract sense? Shouldn't any tests that rely on routes (like... the router tests) be retrying assertions while the router is coming up anyway?

**Comment by crawford** [comment]
The API server comes up "too" quickly. The router was arbitrarily chosen because it is one of the last things to come up. It was a good indication that everything else behaved correctly.

**Comment by wking** [comment]
> Shouldn't any tests that rely on routes (like... the router tests) be retrying assertions while the router is coming up anyway?

I think the e2e tests assume they're running on a working cluster, not one that is in the process of launching itself.

**Comment by deads2k** [comment]
> The API server comes up "too" quickly. The router was arbitrarily chosen because it is one of the last things to come up. It was a good indication that everything else behaved correctly.

Which API server though?  The openshift one?  `oc get --raw /apis/apps.openshift.io/v1` requires dns, networking, and openshift APIs and aggregation before succeeding.

**Comment by abhinavdahiya** [comment]
testing locally, the operator pod is crahlooping:
```
oc -n openshift-ingress-operator logs ingress-operator-b748b79cf-rdbtw
time="2018-10-19T21:29:45Z" level=info msg="Go Version: go1.10.3"
time="2018-10-19T21:29:45Z" level=info msg="Go OS/Arch: linux/amd64"
time="2018-10-19T21:29:45Z" level=info msg="operator-sdk Version: 0.0.6+git"
time="2018-10-19T21:29:46Z" level=info msg="Metrics service ingress-operator created"
time="2018-10-19T21:29:46Z" level=fatal msg="Ensuring default cluster ingress: missing kco-config in configmap"
```

You should be using the `install-config` key from that configmap.

**Comment by ramr** [comment]
@abhinavdahiya  it was fixed a couple of days back with https://github.com/openshift/cluster-ingress-operator/pull/53 ... i'll rebase this and fix the conflicts on monday.

**Comment by ironcladlou** [comment]
I don't think [this condition assertion](https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L156) is valid for a DaemonSet, causing the check to hang.

**Comment by abhinavdahiya** [comment]
> I don't think [this condition assertion](https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L156) is valid for a DaemonSet, causing the check to hang.

cc @smarterclayton

**Comment by smarterclayton** [comment]
What’s the equivalent?  Does a dameonset have a condition we can wait for?

The reason we gate is because of installation lag

**Comment by ironcladlou** [comment]
>What’s the equivalent? Does a dameonset have a condition we can wait for?
>The reason we gate is because of installation lag

Haven't found a way to do it with `oc wait`; the daemonset exposes no conditions. Guess we could compare `status.desiredNumberScheduled` to `status.numberAvailable` from templated output using bash, or something... would that be acceptable? Got a better suggestion?

**Comment by ramr** [comment]
Or if ` oc get ds/router-default -n openshift-ingress -o go-template='{{ne "0" (print .status.numberReady)}}'`  returns `true`, maybe? I couldn't do an int comparison.

**Comment by crawford** [comment]
Tests passed, but it looks like teardown might be having trouble. :/

**Comment by ironcladlou** [comment]
Noooooooooooo

**Pull request merged**